### PR TITLE
Blacklist

### LINKED ISSUE #146: Add blacklist system
We need to standardize this process and avoid future situations where a Hacker that is in our internal blacklist has been accidentally accepted.

My proposal:

1. Add a blacklist structure in the system that can be edited by admins only
    - This blacklist must have fields like: Name | Motive of bann | Date of bann
2. This list can be seen by all HackUPC staff
3. When a blacklisted user registers, the following happens:
    3.1. Show an alert in the application of the blacklisted user so we can reject him/her later
    3.2. Disable the application from being voted while its rejected status is resolved
    3.3. Finally reject the application (notifying the affected user) or uplift the bann

In point 3, sub-points 3.1 and 3.2 will require new code in the system, while 3.3 is better to be manually handled by the directors or the appointed staff.

### PR DESCRIPTION
**Pull request by Ferja**
## What this PR does / Why we need it
Add a blacklist structure in the system that can be edited by admins only, in order to avoid situations where a Hacker that is in the internal blacklist is accidentally accepted.

## Which issue(s) this PR fixes (optional) 
Fixes #146

## Some questions
- [x] I have read the contributing guidelines
- [x] I abide by this repository Code of Conduct
- [x] I understand that my PR won't be merged until GitHub gives a "green light"

## Additional Notes (optional)
Do you want to add anything else? We :heart: to hear your opinions!

### CODE REVIEW

File: app/hackathon_variables.py [hunk 1]
```diff
@@ -88,4 +88,8 @@
 # Enable dubious separate pipeline (disabled by default)
 DUBIOUS_ENABLED = True
 
+
+# Enable blacklist separate pipeline (disabled by default)
+BLACKLIST_ENABLED = True
```
Comment (casassg) [inline app/hackathon_variables.py]: then should we set this to False by default?

File: organizers/views.py [hunk 2]
```diff
@@ -346,3 +368,18 @@ def get_current_tabs(self):
 
     def get_queryset(self):
         return models.Application.objects.filter(status=APP_DUBIOUS)
+
+
+class BlacklistApplicationsListView(TabsViewMixin, IsOrganizerMixin, ExportMixin, SingleTableMixin, FilterView):
```
Comment (Casassarnau) [inline organizers/views.py]: This view comes from IsOrganizerMixin, this means that all organizers can acces here. In order to solve this you have to create a new mixin on User/mixins.py where the function test_func will return true if the request have permisions or if is direcctor

**Review by Casassarnau — CHANGES_REQUESTED** [review]
Could you change this please

File: app/hackathon_variables.py (hunk 1 above)
Comment (Ferja) [reply app/hackathon_variables.py]: Yes, we should

**Review by Casassarnau — APPROVED** [review]
LGTM

**Pull request merged**